dart/lib/src/sentry_exception_factory.dart

```diff
@@ -0,0 +1,37 @@
+import 'protocol/sentry_stack_trace.dart';
+import 'protocol.dart';
+import 'sentry_stack_trace_factory.dart';
+
+/// class to convert Dart Error and exception to SentryException
+class SentryExceptionFactory {
+  final SentryStackTraceFactory _stacktraceFactory;
+
+  const SentryExceptionFactory({
+    SentryStackTraceFactory stacktraceFactory = const SentryStackTraceFactory(),
+  }) : _stacktraceFactory = stacktraceFactory;
+
+  SentryException getSentryException(
+    dynamic exception, {
+    dynamic stackTrace,
+    Mechanism mechanism,
+  }) {
+    if (stackTrace == null && exception is Error) {
+      stackTrace = exception.stackTrace;
```

Comment:
```suggestion
    if (stackTrace == null && exception is Error) {
      stackTrace = exception.stackTrace;
```
```suggestion
    if (exception is Error) {
      stackTrace ??= exception.stackTrace;
```

Reply:
if a `stackTrace` was given and it's not null, we should keep as it is

Reply:
✅